pallets/omnipool/src/provider.rs

```diff
@@ -14,15 +14,16 @@ impl<T: Config> SpotPriceProvider<T::AssetId> for Pallet<T> {
 	fn spot_price(asset_a: T::AssetId, asset_b: T::AssetId) -> Option<Self::Price> {
 		if asset_a == T::HubAssetId::get() {
 			let asset_b = Self::load_asset_state(asset_b).ok()?;
-			asset_b.price()
+			FixedU128::checked_from_rational(asset_b.hub_reserve, asset_b.reserve)
 		} else if asset_b == T::HubAssetId::get() {
 			let asset_a = Self::load_asset_state(asset_a).ok()?;
-			asset_a.price()
```

Comment:
I wonder what we should do with the `price()` function?

Reply:
nothing. it is used throughout omnipool.  It could be used here, but decided to be more explicit here so it is clear.